src/libs/ajax/ajax-common.ts

```diff
@@ -35,6 +35,29 @@ export const withRetryOnError = _.curry((shouldNotRetryFn, wrappedFetch) => asyn
   return wrappedFetch(...args);
 });
 
+const TIMEOUT_DURATION = 10000;
+
+export function makeRequestRetry(request: Function, retriesLeft: number) {
+  return Promise.race([
+    request(),
+    new Promise((_, reject) => setTimeout(() => reject(new Error('Request timed out')), TIMEOUT_DURATION)),
+  ])
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error('Network response was not ok.');
```

Comment:
we probably want to only return on certain status codes, and fail eagerly if we don't get a `retryable` error code